File: 0111_Minimum Depth of Binary Tree/MinimumDepthofBinaryTree_PreOrder.cs
public class Solution {
    private int ans = int.MaxValue;
    public int MinDepth(TreeNode root) {
        if(root == null) return 0;
        Depth(root,1);
        return ans;
    }
    
    private void Depth(TreeNode node, int depth)
    {
        if(node == null) return;
        if(node.left == null && node.right == null)
            ans = Math.Min(ans, depth);
        
        Depth(node.left, depth+1);
        Depth(node.right, depth+1);
    }
}

File: 0111_Minimum Depth of Binary Tree/MinimumDepthofBinaryTree_Preorder.java
class Solution {
    private int ans = Integer.MAX_VALUE;
    public int minDepth(TreeNode root) {
        if(root == null) return 0;
        traverse(root, 1);
        return ans;
    }
    
    private void traverse(TreeNode node, int level) {
        if(node == null) return;
        if(node.left == null && node.right == null) {
            ans = Math.min(ans, level);
        }
        
        traverse(node.left, level+1);
        traverse(node.right, level+1);
    }
}